src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -735,6 +745,13 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
     }
   }
 
+  private def whenGoogleZoneCapacityIssue(throwable: Throwable): Boolean = {
+    throwable match {
+      case t: GoogleJsonResponseException => t.getStatusCode == 429 && t.getDetails.getErrors.asScala.head.getDomain.equalsIgnoreCase("rateLimitExceeded")
```

Comment:
is "rateLimitExceeded" as specific as it gets in the error message? my minor worry is that 429 with "rateLimitExceeded" isn't unique to just google zone issues and we may end up retrying google failures when we shouldn't be. i do acknowledge that google error messages are usually bad and there might not be a better alternative here...

it also seems weird that they're returning a rate limit error when they're the ones out of compute. we're not exceeding any rate limit AFAICT.